File: 3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.cpp
class Solution {
public:
    int maxDistinctElements(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int cnt = 1;
        nums[0] -= k;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] - k > nums[i - 1]) {
                nums[i] -= k;
                cnt++;
            } else if (nums[i] + k <= nums[i - 1]) {
                nums[i] = nums[i - 1];
            } else {
                nums[i] = nums[i - 1] + 1;
                cnt++;
            }
        }
        return cnt;
    }
};

File: 3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.py
class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        nums[0]-=k
        cnt = 1
        for i in range (1, len(nums)):
            if (nums[i] - k > nums[i-1]) :
                nums[i]-=k
                cnt+=1
            elif (nums[i] + k <= nums[i - 1]) :
               nums[i] = nums[i-1]
            else:
                nums[i] = nums[i-1] + 1
                cnt+=1
        return cnt 